numpy/core/src/npymath/npy_math_internal.h.src

```diff
@@ -541,6 +541,22 @@ NPY_INPLACE @type@ npy_cbrt@c@(@type@ x)
  * #C = F, ,L#
  */
 
+@type@ npy_heaviside@c@(@type@ x, @type@ h0)
+{
+    if (npy_isnan(x)) {
+        return (@type@) NPY_NAN;
+    }
+    else if (x == 0) {
+        return h0;
+    }
+    else if (x < 0) {
+        return (@type@) 0.0;
```

Comment:
I like to use integer values for these, but I expect the compiler can also optimize the cast away just as easily for the double values.